### PR TITLE
[JENKINS-41502] Link to upstream trigger pipelines

### PR DESCRIPTION
**Pull request by imeredith**
# Description
Links to upstream build if triggered, from the run header and activity message cell.

To test create 2 pipelines, one the triggers another.

See [JENKINS-41502](https://issues.jenkins-ci.org/browse/JENKINS-41502).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx [hunk 1]
```diff
@@ -155,8 +155,21 @@ class RunDetailsHeader extends Component {
             </div>
         );
 
-        const causeMessage = (run && run.causes.length > 0 && run.causes[run.causes.length - 1].shortDescription) || null;
```
Comment (i386) [inline blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: @imeredith I think you should split this out into a `CauseCell` component that gets reused in both the header and within the `RunMessageCell`.

@kzantow causes have `_class` - should we be making the rendering for different causes an extension point?

Comment (cliffmeyers) [reply blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: Agree, we should absolutely not have teh same code in two places.

Comment (imeredith) [reply blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: Am doing that now

**Comment by imeredith** [comment]
Completion of this is incoming.

**Comment by michaelneale** [comment]
cc @alvarolobato - this is one of the things you brought up some time ago (there are more coming).

**Comment by alvarolobato** [comment]
@michaelneale thanks!!!!

**Pull request merged**